universal_transfer_operator/src/universal_transfer_operator/data_providers/database/google/bigquery.py

```diff
@@ -0,0 +1,200 @@
+    """SnowflakeDataProvider represent all the DataProviders interactions with Snowflake Databases."""
+
+    DEFAULT_SCHEMA = BIGQUERY_SCHEMA
+
+    illegal_column_name_chars: list[str] = ["."]
+    illegal_column_name_chars_replacement: list[str] = ["_"]
+
+    _create_schema_statement: str = "CREATE SCHEMA IF NOT EXISTS {} OPTIONS (location='{}')"
+
+    def __init__(
+        self,
+        dataset: Table,
+        transfer_mode,
+        transfer_params: TransferParameters = attr.field(
+            factory=TransferParameters,
+            converter=lambda val: TransferParameters(**val) if isinstance(val, dict) else val,
+        ),
+    ):
+        self.dataset = dataset
+        self.transfer_params = transfer_params
+        self.transfer_mode = transfer_mode
+        self.transfer_mapping = {}
+        self.LOAD_DATA_NATIVELY_FROM_SOURCE: dict = {}
+        super().__init__(
+            dataset=self.dataset, transfer_mode=self.transfer_mode, transfer_params=self.transfer_params
+        )
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(conn_id="{self.dataset.conn_id})'
+
+    @property
+    def sql_type(self) -> str:
+        return "bigquery"
+
+    @property
+    def hook(self) -> BigQueryHook:
+        """Retrieve Airflow hook to interface with the Snowflake database."""
+        return BigQueryHook(gcp_conn_id=self.dataset.conn_id, use_legacy_sql=False, location=BIGQUERY_SCHEMA_LOCATION)
+
+    @property
+    def sqlalchemy_engine(self) -> Engine:
+        """Return SQAlchemy engine."""
+        uri = self.hook.get_uri()
+        with self.hook.provide_gcp_credential_file_as_context():
+            return create_engine(uri)
+
+    @property
+    def default_metadata(self) -> Metadata:
+        """
+        Fill in default metadata values for table objects addressing snowflake databases
+        """
+        return Metadata(
+            schema=self.DEFAULT_SCHEMA,
+            database=self.hook.project_id,
+        )  # type: ignore
+
+    def read(self):
+        """ ""Read the dataset and write to local reference location"""
+        raise NotImplementedError
+
+    def write(self, source_ref: FileStream):
+        """
+        Write the data from local reference location to the dataset
+
+        :param source_ref: Stream of data to be loaded into snowflake table.
+        """
+        return self.load_file_to_table(
+            input_file=source_ref.actual_file,
+            output_table=self.dataset,
+        )
+
+    # ---------------------------------------------------------
+    # Table metadata
+    # ---------------------------------------------------------
+
+    def schema_exists(self, schema: str) -> bool:
+        """
+        Checks if a dataset exists in the BigQuery
+
+        :param schema: Bigquery namespace
+        """
+        try:
+            self.hook.get_dataset(dataset_id=schema)
+        except GoogleNotFound:
+            return False
```

Comment:
Are there any other exceptions we should be catching here? Also can you write a comment explaining why we catch this exception instead of failing?

Reply:
added details